R/rt_taxo_ranks.R

```diff
@@ -1,26 +1,18 @@
-#' Retrieve the list of taxonomic ranks in TaxRef
+#' Retrieve taxonomic ranks in TaxRef
 #'
-#' @export
-rt_taxo_ranks = function() {
-
-  api_query = rt_GET("taxonomicRanks")
-
-  parse_taxa(api_query)
-}
-
-#' Retrieve a taxonomic rank used in TaxRef
+#' If the function is used without arguments returns the entire list of
+#' taxonomic ranks used in TaxRef.
 #'
-#' @param taxo_id {`character(1)`}\cr{}
-#'                the id of a taxonomic rank in TaxRef see
-#'                [`rt_taxo_ranks()`]
+#' @param taxo_id {`character(1)`, default = `NULL`}\cr{}
+#'                the id of a taxonomic rank in TaxRef.
 #'
 #' @examples
-#' rt_taxo_ranks_id(taxo_id = "VAR")
+#' rt_taxo_ranks(taxo_id = "VAR")
 #'
 #' @export
-rt_taxo_ranks_id = function(taxo_id) {
+rt_taxo_ranks = function(taxo_id = NULL) {
 
-  check_required_arg(taxo_id, "retrieve a specific taxonomic rank using its id")
+  taxo_id = gsub("DUMM", "Dumm", toupper(taxo_id))
```

Comment:
Maybe add `^` and `+` for more security? (to prevent partial matching in other keywords)